### PR TITLE
Rework cursed springs on Icebox

### PR DESCRIPTION
**Pull request by nikita-chekistov**
## About The Pull Request

Cursed springs will now give you virus, that will wabbajack you at the last stage. Same mechanic as the previous cursed spring.

Virus has 9 stages and is pretty slow, but the cure is rezadone, so go good luck searching for carpotoxin.

Stages 1-4 heal player, while 5-8 hurt.  While at stages 1-4, virus can be temporarily stabilized with Ammoniated Mercury (it reverts stages back to 1)

https://github.com/tgstation/tgstation/assets/59139863/ec38363d-00ed-4ebf-a028-c226c0e0eefc

## Why It's Good For The Game

Instant transformation, with no chance to escape _possibly_ turning into a chicken isn't good. And isn't fun either, especially when you step in the hotsprings by accident.
Yeah, rezadone usually is hard to fine, but it's still better than nothing.

## Changelog
:cl: SuperDrish
refactor: Reworked cursed springs on Icebox. Now instead of instantly turning you into random mob, it will give you a special virus.
/:cl:

### CODE REVIEW

**Comment by Microvolnovka19** [comment]
wdym undefined path type 😞

**Comment by Jacquerel** [comment]
> wdym undefined path type 😞

Your new file is not in the dme, so is not being compiled.

**Comment by Helg2** [comment]
The reasoning sounds like a copypaste
Instant `Enter your text here`, with no chance to escape possibly turning into a `Enter your text here` isn't good. And isn't fun either, especially when you step in the `Enter your text here` by accident.

**Comment by Microvolnovka19** [comment]
> So it just won't work on anyone who is virus immune? What's the point of this?

This could be done with one line of code. But how come does the machine will turn into a carp for example? I'd like to stick to non-magical viral background

**Comment by Microvolnovka19** [comment]
yippee-happy

**Comment by Serxule** [comment]
nanite doesnt really make sense, maybe have it be like... FEV? vats/pools of FEV are a thing in fallout and it fits fairly well with the "turn into some random creature"

**Comment by newfren** [comment]
they're supposed to be magic cursed springs right, they should probably work on anything sentient

**Comment by moocowswag** [comment]
cursed springs might aswell be renamed to gene pool or something like that since its not magical anymore.

File: code/datums/diseases/hotsprings.dm [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+				if(affected_mob.getToxLoss())
+					affected_mob.adjustToxLoss(-(stage))
+
+
+			if(SPT_PROB(25, delta_time))
+				var/list/parts = affected_mob.get_damaged_bodyparts(1,1)
+				for(var/obj/item/bodypart/L in parts)
+					if(L.heal_damage(0.25*stage, 0.25*stage, BODYTYPE_ORGANIC))
+						affected_mob.update_damage_overlays()
+				to_chat(affected_mob, span_notice("Your flesh mends by itself."))
```
Comment (Ghommie) [inline code/datums/diseases/hotsprings.dm:44]:
```suggestion
				var/list/parts = affected_mob.get_damaged_bodyparts(1,1)
				var/mended = FALSE
				for(var/obj/item/bodypart/L in parts)
					if(L.heal_damage(0.25*stage, 0.25*stage, BODYTYPE_ORGANIC))
						mended = TRUE
				if(mended)
					affected_mob.update_damage_overlays()
					to_chat(affected_mob, span_notice("Your flesh mends by itself."))
```

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/datums/diseases/hotsprings.dm [hunk 2]
```diff
@@ -0,0 +1,101 @@
+/datum/disease/cursedhotsprings
+	name = "The Misconfigured Nanites"
```
Comment (KittyNoodle) [inline code/datums/diseases/hotsprings.dm:2]: should be just "Misconfigured Nanites"

**Review by Jacquerel — CHANGES_REQUESTED** [review]

**Pull request closed without merging**